AmplifyPlugins/Core/AWSPluginsCoreTests/Model/Decorator/AuthRuleDecorator/ModelWithOwnerAuthAndMultiGroup.swift

```diff
@@ -0,0 +1,242 @@
+        let expectedQueryDocument = """
+        subscription OnCreateOIDCMultiGroupPost {
+          onCreateOIDCMultiGroupPost {
+            id
+            owner
+            title
+            __typename
+          }
+        }
+        """
+        XCTAssertEqual(document.name, "onCreateOIDCMultiGroupPost")
+        XCTAssertEqual(document.stringValue, expectedQueryDocument)
+        XCTAssertTrue(document.variables.isEmpty)
+    }
+
+    func testOnCreateSubscription_InDifferentAppWithAdminsFromMyApp() {
+        let claims = ["username": "user1",
+                      "sub": "123e4567-dead-beef-a456-426614174000",
+                      "https://myapp.com/claims/groups": ["Admins"]] as IdentityClaimsDictionary
+        var documentBuilder = ModelBasedGraphQLDocumentBuilder(modelType: OIDCMultiGroupPost.self,
+                                                               operationType: .subscription)
+        documentBuilder.add(decorator: DirectiveNameDecorator(type: .onCreate))
+        documentBuilder.add(decorator: AuthRuleDecorator(.subscription(.onCreate, claims)))
+        let document = documentBuilder.build()
+        let expectedQueryDocument = """
+        subscription OnCreateOIDCMultiGroupPost {
+          onCreateOIDCMultiGroupPost {
+            id
+            owner
+            title
+            __typename
+          }
+        }
+        """
+        XCTAssertEqual(document.name, "onCreateOIDCMultiGroupPost")
+        XCTAssertEqual(document.stringValue, expectedQueryDocument)
+        XCTAssertTrue(document.variables.isEmpty)
+    }
+
+    func testOnCreateSubscription_InAdminsGroupInDifferentClaim() {
+        let claims = ["username": "user1",
+                      "sub": "123e4567-dead-beef-a456-426614174000",
+                      "https://differentapp.com/claims/groups": ["Admins"]] as IdentityClaimsDictionary
+        var documentBuilder = ModelBasedGraphQLDocumentBuilder(modelType: OIDCMultiGroupPost.self,
+                                                               operationType: .subscription)
+        documentBuilder.add(decorator: DirectiveNameDecorator(type: .onCreate))
+        documentBuilder.add(decorator: AuthRuleDecorator(.subscription(.onCreate, claims)))
+        let document = documentBuilder.build()
+        let expectedQueryDocument = """
+        subscription OnCreateOIDCMultiGroupPost($owner: String!) {
+          onCreateOIDCMultiGroupPost(owner: $owner) {
+            id
+            owner
+            title
+            __typename
+          }
+        }
+        """
+        XCTAssertEqual(document.name, "onCreateOIDCMultiGroupPost")
+        XCTAssertEqual(document.stringValue, expectedQueryDocument)
+        guard let variables = document.variables else {
+            XCTFail("The document doesn't contain variables")
+            return
+        }
+        XCTAssertEqual(variables["owner"] as? String, "123e4567-dead-beef-a456-426614174000")
```

Comment:
```suggestion
        XCTAssertEqual(variables["owner"] as? String, "123e4567-dead-beef-a456-426614174000", "owner should be pesent since group claims does not match auth rule")
```

Reply:
Updated to be slightly more specific.